dbms/src/Storages/Transaction/Collator.cpp

```diff
@@ -337,19 +338,59 @@ const std::array<long_weight, 23> weight_lut_long
        // for default use
        long_weight{0x0, 0x0}};
 
+const std::array<long_weight, 28> weight_lut_long_0900
+    = {long_weight{0x3C013C7B3C000317, 0x000003183CD43C77},
+       long_weight{0x3C073C7B3C000317, 0x0000000003183C80},
+       long_weight{0x3BF53CE03C733C03, 0x0000000000003C7B},
+       long_weight{0x1C0E3D623D673D5E, 0x0000000000003D6E},
+       long_weight{0x3D823D623D863D61, 0x0000000000003D7B},
+       long_weight{0x1C0E3D7C3D863D61, 0x000000003D843D6E},
+       long_weight{0x3D6C3D873D863D61, 0x0000000000003D6E},
+       long_weight{0x3D6E3D7B3D823D62, 0x0000000000003D8B},
+       long_weight{0x3D5B3D683D843D62, 0x0000000000003D86},
+       long_weight{0x1C0E3D6B3D8B3D65, 0x0000000000003D7B},
+       long_weight{0x3D8B3D681C0E3D74, 0x0000000000003D6E},
+       long_weight{0x3D6E3D673D5A3D75, 0x0000000000003D84},
+       long_weight{0x3D6C3D823D5A3D76, 0x0000000000003D6E},
+       long_weight{0x3D5E3D663D6C3D76, 0x0000000000003D84},
+       long_weight{0x1C0E3D6A3D623D77, 0x0000000000003D84},
+       long_weight{0x3D813D663D8B3D79, 0x0000000000003D8B},
+       long_weight{0x1C0E3D743D833D7A, 0x0000000000003D84},
+       long_weight{0x3D633D6E3D8B3D85, 0x0000000000003D8B},
+       long_weight{0xDF0FFB40E82AFB40, 0xF93EFB40CF1AFB40},
+       long_weight{0x06251C8F1C471E33, 0x0000000000001E71},
+       long_weight{0x06251C8F1C471E33, 0x000000001C3F1E71},
+       long_weight{0x020923C5239C2364, 0x23B1239C239C230B},
+       long_weight{0x23250209239C2325, 0x23B1239C230B239C},
+       long_weight{0x000000000000FFFD, 0x0000000000000000},
+       long_weight{0x02091C8F1DB91C3F, 0x00001E331C7A1E71},
+       long_weight{0x1E3302091D321D18, 0x000000001E711CAA},
+       long_weight{0x1E711E711DDD1D77, 0x1E711E711CAA1D77},
+       // for default use
+       long_weight{0x0, 0x0}};
 } // namespace UnicodeCI
 
-class UnicodeCICollator final : public ITiDBCollator
+template <typename T, bool padding>
+class UCACICollator final : public ITiDBCollator
```

Comment:
What is `UCA` mean?

Reply:
[Unicode Collation Algorithm](https://unicode.org/reports/tr10/), the standard of the unicode collation.